controllers/bootstrap/init.go

```diff
@@ -438,8 +438,37 @@ func (b *Bootstrap) CreateOrUpdateFromYaml(yamlContent []byte, alwaysUpdate ...b
 		if len(alwaysUpdate) != 0 {
 			forceUpdate = alwaysUpdate[0]
 		}
-		// TODO: deep merge and update
-		if compareVersion(obj.GetAnnotations()["version"], objInCluster.GetAnnotations()["version"]) || forceUpdate {
+		update := forceUpdate
+
+		// do not compareVersion if the resource is subscription
+		if gvk.Kind == "Subscription" {
+			sub := b.GetSubscription(ctx, obj.GetName(), b.CSData.MasterNs)
+
+			var opt util.OperatorSub
+			opt.Channel = util.GetNestedString(obj, "spec", "channel").(string)
+			opt.InstallPlanApproval = util.GetNestedString(obj, "spec", "installPlanApproval").(string)
+			opt.Name = util.GetNestedString(obj, "spec", "name").(string)
+			opt.Source = util.GetNestedString(obj, "spec", "source").(string)
+			opt.SourceNamespace = util.GetNestedString(obj, "spec", "sourceNamespace").(string)
+			optEnv := util.GetNestedSlice(obj, "spec", "config", "env")
+			var optEnvSlice []map[string]interface{}
+			for _, s := range optEnv {
+				optEnvSlice = append(optEnvSlice, s.(map[string]interface{}))
+			}
+			if len(optEnvSlice) > 0 {
+				opt.InstallScope = optEnvSlice[0]["value"].(string)
+				opt.IsolatedMode = optEnvSlice[1]["value"].(string)
+			}
+
+			if compareSub(sub, opt) {
+				update = true
```

Comment:
Should we add an `else` here? Because subscription of NSS operator is forced updated. the initial value of `update` is `true`

Reply:
Or we could return the value from `compareSub(sub, opt)` and assign it to `update`